scripts/soccer.js

```diff
@@ -586,21 +609,289 @@
+                    fy: relFy
+                });
+            }
+            this.forces = forces;
+        }
+
         // update force applied to robot per time tick
         updateForce(){
+            // Only recalculate here if enough change in robot bearing
             if (this.checkChange() == true){
                 this.calculateForce();
             }
             let forces = this.forces,
                 vector = {},
                 motor;
+            // constantly apply forces as long as motors spin
             for (var i = 0; i < this.numMotors; i++){
                 vector = {x: forces[i].fx, y: forces[i].fy};
                 motor = this.getMotors()[i];
                 Body.applyForce(motor, motor.position, vector);
             }
         }
 
+    }
+
+    class TriBot extends Robot{
+        /**
+         * Bot with three motors, 120 deg from each other. Motors defined in order of left, right, back
+         * @param {*} team team colour
+         * @param {*} x initial position along the x axis
+         * @param {*} y initial position along the y axis
+         */
+        constructor(team, x, y) {
+            super(team);
+            this.createBot(team, x, y, 4, 12);
+        }
+
+        // JQuery and AJAX to fetch SVG file of robot outline
+        parseSVG(name){
+            if (typeof $ !== 'undefined') {
+                // Ensure loading and processing of svg before further processing
+                $.ajaxSetup({async: false});
+                let vertices = [];
+                $.get('../../assets/' + name + '.svg').done(function(data) {
+                    $(data).find('path').each(function(i, path) {
+                        let points = Svg.pathToVertices(path,3);
+                        // Only required if SVG itself is upside down
+                        Vertices.rotate(points,180*degToRad,{x:0,y:0});
+                        vertices.push(points);
+                    });
+                });
+                return vertices;
+            }
+        }
+
+        // Create a square robot with wheel in the centre
+        createBot(team, xPos, yPos, motorWidth, motorHeight){
+
+            // Define a group of parts that won't collide with each other
+            let group = Body.nextGroup(true),
+                // In order of Left, Right, Back motors 
+                motorOffset = [
+                    {x: 17.5, y: 10},
+                    {x: -17.5, y: 10},
+                    {x: 0, y: -17.5}
+                ],
+                bodyColour,
+                motorColour;
+            
+            // Define colours for each team
+            if (team == 'blue'){
+                bodyColour = blue;
+                motorColour = white;
+            } else {
+                bodyColour = yellow;
+                motorColour = black;
+            }
+
+            // Parse the SVG to create vertices
+            let svgName = 'TriBot',
+                vertices = this.parseSVG(svgName);
```

Comment:
Can we store the points in a constant rather than parsing the SVG? `$.ajaxSetup({async: false});` is not good to have